From f30dc3e45d43d9859ffefa828a05fd41d30e4d6b Mon Sep 17 00:00:00 2001
From: Alan Winkowski <walan@marvell.com>
Date: Wed, 21 Jun 2017 20:07:33 +0300
Subject: [PATCH 1725/2241] fix: net: mvpp2x: select correct entry when
 retrieving RSS queue table

- Point to correct RSS queue number when retrieving entries from
  RXQ to RSS table
- No impact in functionality as this code was not in use
- Fixed code will be called from mvpp2x_sysfs

Change-Id: Ie3b2508caa2a7b14034c946cfc70737e2e294607
Signed-off-by: Alan Winkowski <walan@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/40679
Tested-by: iSoC Platform CI <ykjenk@marvell.com>
Reviewed-by: Stefan Chulski <stefanc@marvell.com>
Reviewed-by: Omri Itach <omrii@marvell.com>
---
 drivers/net/ethernet/marvell/mvpp2x/mv_pp2x_hw.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/drivers/net/ethernet/marvell/mvpp2x/mv_pp2x_hw.c b/drivers/net/ethernet/marvell/mvpp2x/mv_pp2x_hw.c
index 516d7ba..b116ff2 100644
--- a/drivers/net/ethernet/marvell/mvpp2x/mv_pp2x_hw.c
+++ b/drivers/net/ethernet/marvell/mvpp2x/mv_pp2x_hw.c
@@ -6298,6 +6298,10 @@ int mv_pp22_rss_tbl_entry_get(struct mv_pp2x_hw *hw,
 		return -EINVAL;
 
 	if (rss->sel == MVPP22_RSS_ACCESS_POINTER) {
+		/* Set index */
+		reg_val |= (rss->u.pointer.rxq_idx <<
+				MVPP22_RSS_IDX_RXQ_NUM_OFF);
+		mv_pp2x_write(hw, MVPP22_RSS_IDX_REG, reg_val);
 		/* Read entry */
 		rss->u.pointer.rss_tbl_ptr =
 			mv_pp2x_read(hw, MVPP22_RSS_RXQ2RSS_TBL_REG) &
-- 
2.7.4

